script/tool/lib/src/version_check_command.dart

```diff
@@ -398,4 +419,40 @@ ${indentation}The first version listed in CHANGELOG.md is $fromChangeLog.
       return null;
     }
   }
+
+  /// Checks whether the current breaking change to [package] should be allowed,
+  /// logging extra information for auditing when allowing unusual cases.
+  bool _validateBreakingChange(RepositoryPackage package) {
+    // Only platform interfaces have breaking change restrictions.
+    if (!package.isPlatformInterface) {
+      return true;
+    }
+
+    if (getBoolArg(_ignorePlatformInterfaceBreaks)) {
+      logWarning(
+          '${indentation}Allowing breaking change to ${package.displayName} '
+          'due to --$_ignorePlatformInterfaceBreaks');
+      return true;
+    }
+
+    if (_getChangeDescription().contains(_breakingChangeJustificationMarker)) {
+      logWarning(
+          '${indentation}Allowing breaking change to ${package.displayName} '
+          'due to "$_breakingChangeJustificationMarker" in the change '
+          'description.');
+      return true;
+    }
+
+    return false;
+  }
+
+  /// Returns the contents of the file pointed to by [_changeDescriptionFile],
+  /// or an empty string if that flag is not provided.
+  String _getChangeDescription() {
+    final String path = getStringArg(_changeDescriptionFile);
+    if (path.isEmpty) {
+      return '';
+    }
+    return packagesDir.fileSystem.file(path).readAsStringSync();
```

Comment:
Check if file exists and `return '';` if it doesn't?

Reply:
I actually didn't do that deliberately; if someone thinks they are passing in a description but has, e.g., typo'd the path, they should get an error rather than have it just silently not work.

I'll make it an explicit ToolExit though, so that it's obvious.